src/url-replacements.js

```diff
@@ -239,6 +242,14 @@ class UrlReplacements {
       return this.getTimingData_('navigationStart',
           'domContentLoadedEventStart');
     });
+
+    // Returns an identifier for the viewer.
+    this.set_('VIEWER', () => {
+      const viewerUrl = viewerFor(this.win_).getParam(VIEWPORT_ORIGIN_PARAM_) ||
```

Comment:
messagingOrigin_ is only set once setMessageDeliverer is called. How do I ensure that it is called before i look for id? 

I am not sure if getReferrerUrl makes sense (that's why the request for comments in original PR). I added it here for the case when origin is not specified. I see that there is a todo in codebase to make origin required. If that is the case, I could remove the referrer part.

/cc @rudygalfi